### PR TITLE
Added cutoff to BaseGridSearch to use any grid search inside evaluate…

### PR DESCRIPTION
**Pull request by aiwalter**
#### What does this implement/fix? Explain your changes.
`BaseGridSearch` did not have an attribute `cutoff` and could therefore not be used as a forecaster in `evaluate` funciton.

### CODE REVIEW

**Comment by aiwalter** [comment]
opened a bug issue at `pmdarima`:
https://github.com/alkaline-ml/pmdarima/issues/423
___
Martin Walter <martin_friedrich.walter@daimler.com>, Mercedes-Benz AG on behalf of Daimler TSS GmbH.
https://github.com/Daimler/daimler-foss/blob/master/LEGAL_IMPRINT.md

**Comment by aiwalter** [comment]
@mloning this contains a hotfix for the `pmdarima` issue.
___
Martin Walter <martin_friedrich.walter@daimler.com>, Mercedes-Benz AG on behalf of Daimler TSS GmbH.
https://github.com/Daimler/daimler-foss/blob/master/LEGAL_IMPRINT.md

File: sktime/forecasting/model_selection/_tune.py [hunk 1]
```diff
@@ -302,6 +302,16 @@ def evaluate_candidates(candidate_params):
...
 
+    @property
+    def cutoff(self):
+        """The time point at which to make forecasts
+
+        Returns
+        -------
+        cutoff : pd.Period, pd.Timestamp, int
+        """
+        return self.best_forecaster_.cutoff
```
Comment (mloning) [inline sktime/forecasting/model_selection/_tune.py:313]: what happens if the algorithm hasn't been fitted yet?

Comment (aiwalter) [reply sktime/forecasting/model_selection/_tune.py:313]: it would throw an error that there is no `best_forecaster_` i think, is this ok like that?

Comment (aiwalter) [reply sktime/forecasting/model_selection/_tune.py:313]: probably best to check `self.check_is_fitted()`

**Review by mloning — APPROVED** [review]
I'll merge this now, but think that we still need to factor out meta algorithms better, rather than implement this for each algorithm.

**Pull request merged**